microservice/package/src/test/java/com/cumulocity/agent/packaging/microservice/impl/MicroserviceDockerClientImplTest.java

```diff
@@ -0,0 +1,293 @@
+package com.cumulocity.agent.packaging.microservice.impl;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.async.ResultCallback;
+import com.github.dockerjava.api.command.*;
+import com.github.dockerjava.api.model.*;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.compress.utils.Sets;
+import org.codehaus.plexus.util.FileUtils;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+@Slf4j
+@ExtendWith(MockitoExtension.class)
+public class MicroserviceDockerClientImplTest {
+
+    private final int TEST_IMAGE_SIZE = 100 * FileUtils.ONE_MB;
+
+    @Mock
+    DockerClient dockerClientMock;
+
+    @InjectMocks
+    MicroserviceDockerClientImpl dockerClient;
+
+    @Rule
```

Comment:
Please, do not mix Junit 4 and Junit 5 --> use Junit 5 `@TempDir ` --- https://www.baeldung.com/junit-5-temporary-directory